Deploy App Service Plan - MLZ Example (#526)
Azure/missionlz: src/bicep/examples/appServicePlan/README.md

### Generate MLZ VAriable File (deploymentVariables.json)

One way to generate the MLZ variable file(deploymentVariables.json) which contains all of the needed values for this examples and others as well is through PowerShell Core and the Azure PowerShell module.  Both PowerShell Core and the Azure PowerShell module are open source projects and avaliable for all major operating systems (Mac, Linux, Windows).

* Get PowerShell Core:  <https://github.com/PowerShell/PowerShell/releases>
* Get Azure PowerShell: <https://docs.microsoft.com/en-us/powershell/azure/install-az-ps>
* Getting Started with Azure PowerShell: <https://docs.microsoft.com/en-us/powershell/azure/get-started-azureps>
* Generate 'deploymentVariables.json': (Get-AzSubscriptionDeployment -Name MLZDeploymentName).outputs | ConvertTo-Json -Depth 10 | Out-File -FilePath .\deploymentVariables.json
* Replace "MLZDeploymentName" with your deployment name:  Browse to 'Subscriptions', to the subscription MLZ was deployed into, and then look at 'Deployments'

> Place the 'deploymentVariables.json' file ./src/bicep/examples folder.  See the sample for reference.

Reviewer: Let's promote the instructions to generate a deployment variables file to the `src/bicep/examples` README and link out to it from this sample. We should keep the `see the sample deployment variables file` bit and update the steps in `Deploying App Service Plan`.

Author: Will come back to this tomorrow night.

Reviewer: When we've settled on where we want it to live and the contents, I'll push in the BASH Azure CLI equivalents too 👍

```plaintext
az login
cd src/bicep
az deployment sub create -n contoso -f mlz.bicep -l eastus --parameters resourcePrefix=contoso
cd examples
az deployment sub show -n contoso --query properties.outputs >> ./deploymentVariables.json
cd appServicePlan
az deployment sub create -n deployAppServicePlan -f appService.bicep -l eastus
```

Author: Documentation has been updated as requested.  Thank you Glenn!

# App Service Plan Example

> This example deploys an App Service Plan (AKA: Web Server Cluster) to support simple web accessible linux docker containers.  It also optionally supports the use of dynamic(up and down) scale settings based on CPU percentage up to a max of 10 compute instances.

Reviewer 2: ```suggestion
This example deploys an App Service Plan (AKA: Web Server Cluster) to support simple web accessible linux docker containers.  It also optionally supports the use of dynamic (up and down) scale settings based on CPU percentage up to a max of 10 compute instances.
```

The subscription and resource group can be changed by providing the resource group name (Param: targetResourceGroup) and ensuring that the Azure context is set the proper subscription.  

## Pre-requisites

1. A Mission LZ deployment (a deployment of mlz.bicep)
2. The outputs from a deployment of mlz.bicep (./src/bicep/examples/deploymentVariables.json).  

See below for information on how to create the appropriate deployment variables file for use with this template.

### Template Parameters

Template Parameters Name | Description
-----------------------| -----------
appServicePlanName | The name of the App Service Plan.  If not specified, the name will default to the MLZ default naming pattern.  
targetResourceGroup | The name of the resource group where the App Service Plan will be deployed.   If not specified, the resource group name will default to the shared services MLZ resource group name and subscription.

> enableAutoScale | A true/false value that determines if dyname auto scale is enabled.  If set to "true", dyanmic auto scale is enabled up to a maximum of 10 compute instances based on CPU percentage exceeding 70% for 10 minutes.   Will also scale in if CPU percentage is below 30% for 10 minutes.  If set to "false", the App Service Plan will statically maintain two compute instances indefinitely.

Reviewer 2: ```suggestion
enableAutoScale | A true/false value that determines if dynamic auto scale is enabled.  If set to "true", dynamic auto scale is enabled up to a maximum of 10 compute instances based on CPU percentage exceeding 70% for 10 minutes.   Will also scale down if CPU percentage is below 30% for 10 minutes.  If set to "false", the App Service Plan will statically maintain two compute instances indefinitely.
```

> ### Generate MLZ VAriable File (deploymentVariables.json)

Reviewer 2: ```suggestion
### Generate MLZ Variable File (deploymentVariables.json)
```
